### ISSUE TITLE
Install issue

### ISSUE THREAD

**Issue by gugaiz** [deliverybot/deliverybot#54]
I just cloned the repo and ran
```
yarn install
yarn start
```
as the readme suggests, but I got when I ran `yarn start`

```
core:       [14:51:22] Starting compilation in watch mode...
.:          internal/modules/cjs/loader.js:329
.:                throw err;
.:                ^
.:          Error: Cannot find module '/var/www/deliverybot/node_modules/@deliverybot/deploybot/lib/index.js'. Please 
...
.:          error Command failed with exit code 1.
.:          info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Any suggestions?

**Comment by colinjfw** [comment]
That most likely means that `@deliverybot/deploybot` wasn't loaded locally or built. It looks like it's defined here in the yarn workspaces configuration https://github.com/deliverybot/deliverybot/blob/master/package.json#L9.

**Comment by gugaiz** [comment]
Thanks for your answer @colinjfw. I got that package when I ran `yarn install` but there is no `lib` directory inside
```
$ ls node_modules/@deliverybot/deploybot
jest.config.js  node_modules  package.json  README.md  src  test  tsconfig.json
```

I realized that there is a missing step on the README.. `yarn build`

**Comment by gugaiz** [comment]
Now it seems I am missing the secrets but not sure where to place them

```
app:        Synced partials to lib/views/partials.js
.:          /var/www/deliverybot/node_modules/@octokit/webhooks/index.js:11
.:              throw new Error('options.secret required')
.:              ^
.:          Error: options.secret required
.:              at new createWebhooksApi (/var/www/deliverybot/node_modules/@octokit/webhooks/index.js:11:11)
.:              at Object.probot (/var/www/deliverybot/packages/core/lib/probot.js:64:17)
.:              at Object.load (/var/www/deliverybot/packages/core/lib/app.js:35:26)
.:              at Object.<anonymous> (/var/www/deliverybot/packages/run/lib/index.js:39:25)
.:              at Module._compile (internal/modules/cjs/loader.js:1137:30)
.:              at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
.:              at Module.load (internal/modules/cjs/loader.js:985:32)
.:              at Function.Module._load (internal/modules/cjs/loader.js:878:14)
.:              at Module.require (internal/modules/cjs/loader.js:1025:19)
.:              at require (internal/modules/cjs/helpers.js:72:18)
.:          error Command failed with exit code 1.
.:          info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
.:          server exited with code 1
run:        webpack is watching the files…
```

**Comment by denschu** [comment]
@gugaiz I think you need to create a GitHub app first: https://docs.github.com/en/developers/apps/creating-a-github-app
After that you can fill out the values for the variables in the `.env` file.
I followed this steps: https://probot.github.io/docs/development/#manually-configuring-a-github-app

 @colinjfw Can you help us on how to configure the GitHub app to be able to run deliverybot in our local development environment? Especially it would be interesting to know which values we need to provide for the "User authorization callback URL". An empty value leads to that message: "This GitHub App must be configured with a callback URL".

Thanks in advance :-)

**Comment by denschu** [comment]
Ok, for local development this issue is helpful: https://github.com/deliverybot/deliverybot/issues/44 
The `BASE_URL` needs to be set.
Next step would be to get it up and running with a cloud provider, e.g. AWS (Beanstalk) :-) Anyone successful with that and can help us?
I saw that `yarn start` does some proxy handling with Smee, so that seems not to be the right command for production.

**Comment by colinjfw** [comment]
> Can you help us on how to configure the GitHub app to be able to run deliverybot in our local development environment?

Yes definitely, so the most important part here is the `.env.example` file in the root of the directory. First step is to copy that file to `.env` and then fill in all of the environment variables. This will entail creating a GitHub App with the same permissions as described as in the file `app.yml`.

**Comment by denschu** [comment]
@colinjfw Thanks for your reply 👍  Can you also give us some insights on how to configure, build and run deliverybot for the production deployment?

**Comment by ramene** [comment]
There's a similar conversation on [spectrum.chat/deliverybot](https://spectrum.chat/deliverybot/general/delierybot-is-awesome-so-sad-to-see-it-shut-down~3a5195bc-7314-4ea7-82b4-14513b511d12)

TL;DR. The packages need to be built first.

Run `yarn build` _prior_ to `yarn server`

I'll submit a PR tonight, with a multistage Dockerfile and accompanying `kubernetes.yml` that can use to deploy locally or any cloud provider

**Comment by ramene** [comment]
> @gugaiz I think you need to create a GitHub app first: https://docs.github.com/en/developers/apps/creating-a-github-app
> After that you can fill out the values for the variables in the `.env` file.
> I followed this steps: https://probot.github.io/docs/development/#manually-configuring-a-github-app
> 
> @colinjfw Can you help us on how to configure the GitHub app to be able to run deliverybot in our local development environment? Especially it would be interesting to know which values we need to provide for the "User authorization callback URL". An empty value leads to that message: "This GitHub App must be configured with a callback URL".
> 
> Thanks in advance :-)

@gugaiz `/login/cb`

https://github.com/deliverybot/deliverybot/blob/041a34d3d089481031db28f4ba76aecd2d21f124/packages/app/src/apps/auth.ts#L148
